src/emulator/download.ts

```diff
@@ -4,29 +4,55 @@ import * as ProgressBar from "progress";
 
 import * as FirebaseError from "../error";
 import * as utils from "../utils";
-import { Emulators } from "./types";
+import { Emulators, JavaEmulatorDetails } from "./types";
 import * as javaEmulators from "../serve/javaEmulators";
 import * as tmp from "tmp";
 import * as fs from "fs-extra";
+import * as path from "path";
 
 tmp.setGracefulCleanup();
 
 type DownloadableEmulator = Emulators.FIRESTORE | Emulators.DATABASE;
 
 module.exports = (name: DownloadableEmulator) => {
   const emulator = javaEmulators.get(name);
-  utils.logLabeledBullet(name, "downloading emulator...");
+  utils.logLabeledBullet(name, `downloading ${path.basename(emulator.localPath)}...`);
   fs.ensureDirSync(emulator.cacheDir);
-  return downloadToTmp(emulator.remoteUrl).then((tmpfile) =>
-    validateSize(tmpfile, emulator.expectedSize)
-      .then(() => validateChecksum(tmpfile, emulator.expectedChecksum))
-      .then(() => {
-        fs.copySync(tmpfile, emulator.localPath);
-        fs.chmodSync(emulator.localPath, 0o755);
-      })
-  );
+
+  return downloadToTmp(emulator.remoteUrl)
```

Comment:
(totally optional, but...) could you pretty please async/await this? it'd look so good!

```typescript
const tmpFile = await downloadToTmp(emulator.removeUrl);
await validateSize(tmpFile, emulator.expectedSize)
await validateChecksum(tmpFile, emulator.expectedChecksum)
fs.copy
fs.chmod
removeOldJars(emulator)
```